go-controller/pkg/ovn/master.go

```diff
@@ -908,5 +994,44 @@ func (oc *Controller) syncNodes(nodes []interface{}) {
 		if err := oc.deleteNode(nodeName, nodeSubnets.hostSubnet, nodeSubnets.joinSubnet); err != nil {
 			klog.Error(err)
 		}
+		//remove the node from the chassis map so we don't delete it twice
+		delete(chassisMap, nodeName)
+	}
+
+	for nodeName, chassisName := range chassisMap {
+		if chassisName != "" {
+			_, stderr, err = util.RunOVNSbctl("--if-exist", "chassis-del", chassisName)
+			if err != nil {
+				klog.Errorf("Failed to delete chassis with name %s for logical switch %s: stderr: %q, error: %v",
+					chassisName, nodeName, stderr, err)
+			}
+		}
+	}
+}
+
+func (oc *Controller) unmarshalChassisDataIntoMap(chData []byte) (map[string]string, error) {
+	//map of node name to chassis name
+	chassisMap := make(map[string]string)
+	var mapUnmarshal map[string][][]string
+
+	if len(chData) == 0 {
+		return chassisMap, nil
+	}
+
+	err := json.Unmarshal(chData, &mapUnmarshal)
```

Comment:
this will return an error, because "headings" is not a [][]string, its just a []string. To make this less fragile (as you suggested in your other comment) and even cleaner you can just use a struct:

```
	chassisMap := make(map[string]string)
        chData := []byte(`{"data":[["1553f2c3-6c71-4bf1-b67d-d318d19e5101","ovn-control-plane"]],"headings":["name","hostname"]}`)
	type chassisOut struct {
		Data [][]string
		Headings []string
        }
	var mapUnmarshal chassisOut
	err := json.Unmarshal(chData, &mapUnmarshal)
	fmt.Println("error: ", err, "data: ", mapUnmarshal)
	for _, entry := range mapUnmarshal.Data {
		if len(entry) < 2 || entry[0] == "" || entry[1] == "" {
			 continue
		}
		chassisMap[entry[0]] = entry[1]
	}
        fmt.Println(chassisMap)
```
You can even omit the Headings field, and unmarshal will just ignore it and return no error. You also don't need to check if data exists anymore by using this method.

Reply:
Good catch. I overlooked the fact that headings is not a [][]string. I like this much better than the first implementation. Updated the PR.